src/protections/ProtectionManager.ts

```diff
@@ -100,6 +102,9 @@ export class ProtectionManager {
             protection.settings[key].setValue(value);
         }
         if (protection.enabled) {
+            if (protection.name === "NsfwProtection") {
+                (protection as NsfwProtection).initialize()
+            }
```

Comment:
an alternative would be to add `initialize()` to the interface on all of the protections, then have the others do nothing.

Reply:
Yeah I chose this way since it seemed less desirable to add a field to the abstract class that only this protection used.